api/groups.go

```diff
@@ -0,0 +1,17 @@
+package api
+
+import (
+	"context"
+)
+
+type GroupsClient interface {
+	AddGroupMember(ctx context.Context, groupObjectID string, memberObjectID string) error
+	RemoveGroupMember(ctx context.Context, groupObjectID, memberObjectID string) error
+	GetGroup(ctx context.Context, objectID string) (result ADGroup, err error)
+	ListGroups(ctx context.Context, filter string) (result []ADGroup, err error)
+}
+
+type ADGroup struct {
+	ID          string
+	DisplayName string
+}
```

Comment:
This interface is returning `ADGroup`, which seems specific to AAD. Is this client also not being used for msgraph?

Reply:
I don't think this is specific to AAD. I created this struct (as opposed to moving it from the main package) so I could create the interface that abstracts the differences between AAD and msgraph. If I recall correctly, the interface did exist, but it just matched the AAD client's interface but I changed it so it was abstracted away from the AAD implementation (which also meant I needed to add this `ADGroup` struct). Unless I'm misunderstanding your question?

Reply:
Can we change the name of ADGroup, then?